internal/collector/event.go

```diff
@@ -46,3 +51,31 @@ func (collector *EventCollector) Describe(ch chan<- *prometheus.Desc) {
 func (collector *EventCollector) Collect(ch chan<- prometheus.Metric) {
 	collector.eventsTotal.Collect(ch)
 }
+
+// WithInformerFactory adds a informers.SharedInformerFactory to the collector.
+func (collector *EventCollector) WithInformerFactory(factory informers.SharedInformerFactory) {
+	collector.informerFactory = factory
+}
+
+// Run starts updating EventCollector metrics.
+func (collector *EventCollector) Run(stopCh <-chan struct{}) {
+	eventsTotalInformer := collector.informerFactory.Core().V1().Events().Informer()
+	eventsTotalInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			collector.increaseEventsTotal(obj.(*v1.Event))
+		},
+		UpdateFunc: func(obj, _ interface{}) {
+			collector.increaseEventsTotal(obj.(*v1.Event))
+		},
+	})
+	go eventsTotalInformer.Run(stopCh)
+}
+
+func (collector *EventCollector) increaseEventsTotal(event *v1.Event) {
+	collector.eventsTotal.With(prometheus.Labels{
+		"type":                      event.Type,
+		"involved_object_namespace": event.InvolvedObject.Namespace,
```

Comment:
Curious if we want to use `CurryWith` here for the namespace https://github.com/prometheus/client_golang/blob/a200f1930c10bce54c6770a8fd115857f8696574/prometheus/counter.go#L264-L270

Reply:
Also did we say we don't care about the namespace of the event right?

Reply:
> Curious if we want to use CurryWith here for the namespace

If the purpose is to control the cardinality, we will already be able to do so by applying namespace filtering on the informer. I don't feel that this is the right place to do this filtering and would rather do it at the informer level.

Reply:
> Also did we say we don't care about the namespace of the event right?

We did, I only included the namespace of the involved object as we previously planned. The only difference I noticed so far concerns Events created by Nodes. Since node objects are not namespaced their involved_object_namespace label will be empty and their Events will be created under the `default` namespace.